src/game/state/availablePointsForInsect.js

```diff
@@ -61,5 +61,25 @@ export const availablePointsForInsect = {
       currentPoints = subtract(newPoints, visited);
     }
     return currentPoints;
+  },
+  grasshopper: ({G, currentInsect}) => {
+    const insectsPoints = G.insects.map(({point}) => point);
+    const neighborInsectPoints = getNeighbors(currentInsect.point)
+      .filter(n =>  insectsPoints.some(i => isSame(n)(i)));
```

Comment:
I think this filter can be removed since it is implemented on :77?

Reply:
The grasshopper must jump over at least one insect, so I need to know which neighbor cells that is occupied by an insect to know which vectors that are allowed.

The algorithm is:
1. Find all neighbor insects
2. For each neighbor insect, calculate the jump vector
3. Find the first empty cell using the current insects position and the jump vector

If the filter is removed, then we would get any unoccupied neighbor cells as  a valid point.

Reply:
Of course, you are right.